### PR TITLE
Disable valgrind for most -jac tests

### LINKED ISSUE #15142: Skip Valgrind testing for PETSc Jacobian tests if a non *-jac version exists
## Reason
We do not exclude a lot of jacobian tests from valgrind testing *-jac tests in TM for example. That seems like an unnecessary burden, as those test have a non-jac variant that already gets valgrind tested. So we are basically just valgrind testing the petsc jacobian checker code.

## Design
Go over the tests and set `valgrind = none` for the `*-jac` tests mentioned above.

## Impact
Faster testing, fewer timeouts.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #15142

### CODE REVIEW

**Comment by dschwen** [comment]
@WilkAndy I haven't touched porous_flow, which seems to have a ton of `PetscJacobianTester` tests.

**Comment by dschwen** [comment]
The failing tests were added by @tophmatthews but due to copy paste errors they were never run. Could you take a quick look, Thopher, if it the tolerances need adjusting?

**Comment by WilkAndy** [comment]
Regarding the PorousFlow tests - it's probably overkill to valgrind them, since i think the valgrind test coverage is already sufficient, BUT, unless you guys are seriously running out of compute resources, please DO valgrind them, just in case.  Thanks @dschwen .

**Comment by dschwen** [comment]
@wilkandy, the jacobian tests I un-valgrinded were also running as Exodiff tests. The same resudual abd Jacobian evaluatuins were performed, the only difference is that PetscJacobianTestster adds additional overhead to compare to the finite differenced Jacobian. The only added value to valgrind testing would be to over and over and over test the PETSc finite differencing code.

**Comment by tophmatthews** [comment]
> The failing tests were added by @tophmatthews but due to copy paste errors they were never run. Could you take a quick look, Thopher, if it the tolerances need adjusting?

@dschwen No problem Dhaniel

**Comment by tophmatthews** [comment]
Wouldn't this make more sense to change `PetscJacobianTester` to have defaults for `method=opt`, `valgrind=none`, `petsc_version>=3.9.4`? Maybe even `cli_args = 'Outputs/exodus=false Outputs/csv=false` At some point I was going to remove `difference_tol` altogether and adding in snes_test_err as an option, but that was lost in the list of todos.

**Comment by tophmatthews** [comment]
The modules failure is due to duplicated valgrind param

```
/opt/civet/build_0/moose/modules/tensor_mechanics/test/tests/ad_finite_strain_jacobian/tests:78: duplicate parameter "/Tests/3d_bar-jac/valgrind"
/opt/civet/build_0/moose/modules/tensor_mechanics/test/tests/ad_finite_strain_jacobian/tests:94: duplicate parameter "/Tests/3d_bar_Bbar-jac/valgrind"
/opt/civet/build_0/moose/modules/tensor_mechanics/test/tests/ad_2D_geometries/tests:120: duplicate parameter "/Tests/3D_RZ_finitestrain-jac/valgrind"
```

**Comment by tophmatthews** [comment]
ad_2d_diffusion needs to be loosened:
```
kernels/ad_2d_diffusion.testdirichlet-jac:   ||J - Jfd||_F/||J||_F = 2.81068e-09, ||J - Jfd||_F = 1.20346e-08
```
difference_tol = 1e-7

This is why difference_tol means nothing and should be dropped from the check!

**Comment by dschwen** [comment]
Thorry, bit on my thounghe.

**Comment by dschwen** [comment]
> Wouldn't this make more sense to change `PetscJacobianTester` to have defaults for `method=opt`, `valgrind=none`, `petsc_version>=3.9.4`? Maybe even `cli_args = 'Outputs/exodus=false`

Yeah, I agree, and it crossed my mind. I haven't seen any tester override defaults from the class it inherits. let me check if that s possible without too much pain. And the other consideration was that I only wanted to disable valgrind testing if the input file is also run using a different tester. But I guess we could just pull the bandaid and put the onus on the developer to make sure valgrind test coverage is given by non-jacobian tests.

**Comment by dschwen** [comment]
Never mind, got it.

File: python/TestHarness/testers/PetscJacobianTester.py [hunk 1]
```diff
@@ -26,6 +26,12 @@ def validParams():
                                           "at every non-linear iteration of every time step. This is only "
                                           "relevant for petsc versions >= 3.9.")
         params.addParam('turn_off_exodus_output', True, "Whether to set exodus=false in Outputs")
+
+        # override default values
+        params.valid['valgrind'] = 'none'
+        params.valid['petsc_version'] = '>=3.9.4'
+        params.valid['method'] = 'opt'
+
```
Comment (tophmatthews) [inline python/TestHarness/testers/PetscJacobianTester.py:34]: I probably know what you're going to say...but what about removing `difference_tol` and adding in `snes_test_err` while you're tooling around in here?

**Comment by dschwen** [comment]
> Never mind, got it.

Apparently I haven't "got it" :-D

File: python/TestHarness/testers/PetscJacobianTester.py (hunk 1 above)
Comment (dschwen) [reply python/TestHarness/testers/PetscJacobianTester.py:34]: I haven't followed the `snes_test_err` discussion. Why is that approach better?

Comment (tophmatthews) [reply python/TestHarness/testers/PetscJacobianTester.py:34]: It's not better, it's two different things, sorry for conflating them. `difference_tol` doesn't contain the jacobian check information. Right now `difference_tol` is just set to some number above the `||J - Jfd||_F` by guess and check (so really you could leave it but give it a really high default tolerance.

`snes_test_err` sometimes needs to be adjusted to get `ratio_tol` low. This is now set through cli_args, but feels like it should be a tester option.

 @lindsayad can help clarify.

Comment (dschwen) [reply python/TestHarness/testers/PetscJacobianTester.py:34]: Oh, ok. Is `snes_test_err` the finite differencing parameter?

Comment (lindsayad) [reply python/TestHarness/testers/PetscJacobianTester.py:34]: > Oh, ok. Is `snes_test_err` the finite differencing parameter?

Yes. Yea `difference_tol` is highly subject to the scaling of your system.

**Review by permcody — APPROVED** [review]

**Pull request merged**